apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+            <input
+              type="checkbox"
+              value={this.props.value}
+              checked={this.props.selected}
+              ref="isCheckedInput"
+              onChange={this.handleChange}
+              style={{marginRight: 5}}
+            />
+            {this.props.text}
+          </label>
+        </div>
+      );
+    }
+  });
+
+  const FilterGroup = React.createClass({
+    propTypes: {
+      name: React.PropTypes.string,
+      text: React.PropTypes.string,
+      filterEntries: React.PropTypes.array,
+      selection: React.PropTypes.array,
+      onUserInput: React.PropTypes.func
+    },
+
+    render() {
+      return (
+        <div style={{paddingTop: 20, paddingRight: 40}}>
+          <div style={{fontFamily: '"Gotham 5r", sans-serif', borderBottom: 'solid grey 1px'}}>
+            {this.props.text}
+          </div>
+          {this.props.filterEntries.map(item => <FilterChoice groupName={this.props.name} name={item.name} text={item.text} selected={this.props.selection && this.props.selection.includes(item.name)} onUserInput={this.props.onUserInput} key={item.name}/>)}
+        </div>
+      );
+    }
+  });
+
+  const FilterSet = React.createClass({
+    propTypes: {
+      filterGroups: React.PropTypes.array,
+      onUserInput: React.PropTypes.func,
+      selection: React.PropTypes.object
+    },
+
+    render() {
+      return (
+        <div>
+          <div className="col-20">
+            <div style={{fontSize: 16}}>
+              Filter By
+            </div>
+            {this.props.filterGroups.map(item => <FilterGroup name={item.name} text={item.text} filterEntries={item.entries} onUserInput={this.props.onUserInput} selection={this.props.selection[item.name]} key={item.name}/>)}
+          </div>
+        </div>
+      );
+    }
+  });
+
+  // For a string of tags, generate a string of their friendly names.
+  function getTagString(prefix, tagString) {
+    if (!tagString) {
+      return "";
+    }
+
+    var tagToString = {
+      length_1_hour: "One hour",
+      subject_english: "English",
+      subject_history: "History",
+      teacher_experience_expert: "Expert",
+      teacher_experience_beginner: "Beginner",
+      student_experience_expert: "Expert",
+      student_experience_beginner: "Beginner",
+      activity_type_programming: "Programming tutorial",
+      grade_pre: "Pre",
+      "grade_2-5": "2-5",
+      "grade_6-8": "6-8",
+      programming_language_javascript: "JavaScript",
+      programming_language_c: "C",
+    };
+
+    return tagString.split(',').map(tag => tagToString[`${prefix}_${tag}`]).join(', ');
+  }
+
+  var TutorialDetail = React.createClass({
+    propTypes: {
+      showing: React.PropTypes.bool,
+      item: React.PropTypes.object,
+      closeClicked: React.PropTypes.func,
+    },
+
+    render: function () {
+      if (!this.props.showing) {
+        // Disable body scrolling.
+        $('body').css('overflow', 'auto');
```

Comment:
Fully agreed.  Unfortunately this was my best idea for now, unless we want to generate the body tag itself via the same React chain?